GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -0,0 +1,120 @@
+﻿using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.Common.Maintenance
+{
+    public class LooseObjectsStep : GitMaintenanceStep
+    {
+        public const string LooseObjects = "loose-objects.history";
+        private readonly string looseObjectsHistoryPath = null;
+        private readonly bool ignoreTimeRestriction = false;
+
+        public LooseObjectsStep(GVFSContext context, bool requireCacheLock, bool ignoreTimeRestriction)
+            : this(context, requireCacheLock)
+        {
+            this.ignoreTimeRestriction = ignoreTimeRestriction;
+        }
+
+        public LooseObjectsStep(GVFSContext context, bool requireCacheLock)
+            : base(context, gitObjects: null, requireObjectCacheLock: requireCacheLock)
+        {
+            this.looseObjectsHistoryPath = Path.Combine(this.Context.Enlistment.GitObjectsRoot, "info", LooseObjects);         
+        }
+
+        public override string Area => "LooseObjectsStep";
+
+        public int CountLooseObjects()
+        {
+            int count = 0;
+            int pathLength = this.Context.Enlistment.GitObjectsRoot.Length;
+            
+            foreach (string directory in this.Context.FileSystem.EnumerateDirectories(this.Context.Enlistment.GitObjectsRoot))
+            {
+                // Check if the directory is 2 letter HEX
+                if (System.Text.RegularExpressions.Regex.IsMatch(directory, @"[0-9a-fA-F]{2}"))
+                {                    
+                    count += this.Context.FileSystem.GetFiles(Path.Combine(this.Context.Enlistment.GitObjectsRoot, directory), "*").Count();
+                }
+            }
+
+            return count;
+        }
+
+        protected override void PerformMaintenance()
+        {
+            using (ITracer activity = this.Context.Tracer.StartActivity("LooseObjectsStep", EventLevel.Informational, Keywords.Telemetry, metadata: null))
+            {
+                try
+                {
+                    if (!this.ignoreTimeRestriction && !this.EnoughTimeBetweenRuns())
+                    {
+                        activity.RelatedWarning("Skipping step due to not enough time between runs");
+                        return;
+                    }
+
+                    // Verify no git processes are running
+                    IEnumerable<int> processIds = this.RunningGitProcessIds();

```

Comment:
Should this method return a `bool` instead?

Also, better to use `.Any()` instead of `.Count() > 0` on an `IEnumerable`.